department-of-veterans-affairs/vets-website
# 11817 Augments Announcements w/ once-per-session option

Closes department-of-veterans-affairs/va.gov-cms/issues/11817

## Summary

- This PR changes the announcement code to have a 3rd time of behavior, allowing announcements to be seen once-per-session.
- I've altered existing code, copying it's established patterns to augment the behavior. This required changing what was once a boolean field to an Enum field.
- Public Websites. To the best of my knowledge my team owns this component's maintenance. 


## Related issue(s)
- department-of-veterans-affairs/va.gov-cms/issues/11817
- New Homepage epic, link not available as I've not able to access ZenHub today.


## Testing done

- Previously there was a boolean value to determine if a modal announcement was shown everytime or just once.
- See Below of Testing / QA Steps
- *Describe the tests completed and the results*


## QA Steps 
In order to test the new functionality of the Announcement Framework, I've temporarily configured 3 test modals to exhibit the 3 behaviors modes.

### Show Once 
1. Visit `/test-announcement--once` 
2. See the modal popup.
3. Dismiss the modal.
4. Refresh page.
5. Note the absence of the modal. 
6. Check Dev Tools > Application > Local Storage and note the presence of "testing-announcements-once-per-session".

### Show Every time
1. Visit `/test-announcement--every-time`
2. See the modal popup.
3. Dismiss the modal.
3. Refresh page.
4. See the modal popup again. 
5. Observe the that no new entry has been added to localStorage or sessionStorage

### Show Once Per Session 
1. Visit `/test-announcement--once-per-session` 
2. Observe the modal popup. 
3. Dismiss the Modal.
4. Check Dev Tools > Application > Session Storage and note the presence of "testing-announcements-one-per-session".
5. Refresh the page and observe no modal is shown.
6. Close tab and revisit path in Step 1. 
7. Observe that modal is once again present.

## What areas of the site does it impact?
This component effects every page that desires to have a modal announcement present. At present, this only occurs when redirected from https://explore.va.gov, though it will soon be removed. Soon, this change will be utilized to allow for once-per-session display of a modal to allow users to visit the new Homepage Experience.

## Acceptance criteria

- [n/a]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [n/a]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [n/a]  I added a screenshot of the developed feature

## Requested Feedback
Assure the modal announcements 3 behaviors are working as expected by following the QA Steps provided above. 

## NOTE:
After behavior is QA is approved, either an change will be required to remove the test / demonstration config that was required to demonstrate the behaviors for the QA Steps listed above.
